itest/litd_custom_channels_test.go

```diff
@@ -3434,3 +3432,222 @@ func runCustomChannelsHtlcForceClose(ctxb context.Context, t *harnessTest,
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(charlie, dave))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(dave, charlie))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(dave, yara))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(yara, dave))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(erin, fabia))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(fabia, erin))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(charlie, erin))
+
+	logBalance(t.t, nodes, assetID, "initial")
+
+	// We now deplete the channel between Erin and Fabia by moving all
+	// assets to Fabia.
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, erinFundingAmount, assetID, fn.None[int64](),
+	)
+	logBalance(t.t, nodes, assetID, "after moving assets to Fabia")
+
+	// Test case 1: We cannot keysend more assets from Erin to Fabia.
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, 1, assetID, fn.None[int64](),
+		withFailure(lnrpc.Payment_FAILED, failureNoBalance),
+	)
+
+	// Test case 2: We cannot pay an invoice from Charlie to Fabia.
+	invoiceResp := createAssetInvoice(t.t, erin, fabia, 123, assetID)
+	payInvoiceWithSatoshi(
+		t.t, charlie, invoiceResp,
+		withFailure(lnrpc.Payment_FAILED, failureNoRoute),
+	)
+
+	// Test case 3: We now create an asset buy order for a normal amount of
+	// assets. We then "fake" an invoice referencing that buy order that
+	// is for an amount that is too small to be paid with a single asset
+	// unit. This should be handled gracefully and not lead to a crash.
+	// Ideally such an invoice shouldn't be created in the first place, but
+	// we want to make sure that the system doesn't crash in this case.
+	numUnits := uint64(10)
+	buyOrderResp, err := fabiaTap.RfqClient.AddAssetBuyOrder(
+		ctxb, &rfqrpc.AddAssetBuyOrderRequest{
+			AssetSpecifier: &rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetId{
+					AssetId: assetID,
+				},
+			},
+			AssetMaxAmt: numUnits,
+			Expiry: uint64(
+				time.Now().Add(time.Hour).Unix(),
+			),
+			PeerPubKey:     erin.PubKey[:],
+			TimeoutSeconds: 10,
+		},
+	)
+	require.NoError(t.t, err)
+
+	quoteResp := buyOrderResp.Response
+	quote, ok := quoteResp.(*rfqrpc.AddAssetBuyOrderResponse_AcceptedQuote)
+	require.True(t.t, ok)
+
+	// We calculate the milli-satoshi amount one below the equivalent of a
+	// single asset unit.
+	rate, err := oraclerpc.UnmarshalFixedPoint(&oraclerpc.FixedPoint{
+		Coefficient: quote.AcceptedQuote.AskAssetRate.Coefficient,
+		Scale:       quote.AcceptedQuote.AskAssetRate.Scale,
+	})
+	require.NoError(t.t, err)
+
+	oneUnit := rfqmath.NewBigIntFixedPoint(1, 0)
+	oneUnitMilliSat := rfqmath.UnitsToMilliSatoshi(oneUnit, *rate)
+
+	t.Logf("Got quote for %v asset units per BTC", rate)
+	msatPerUnit := float64(oneUnitMilliSat) / float64(1)
+	t.Logf("Got quote for %v asset units at %3f msat/unit from peer %s "+
+		"with SCID %d", numUnits, msatPerUnit, erin.PubKeyStr,
+		quote.AcceptedQuote.Scid)
```

Comment:
In general, I think it would be better if log messages were written with args at the end:
```
t.Logf("Got quote from peer (asset_units=%v, msat_to_asset_units=%3f, peer=%s, "+
		"scid=%d)", numUnits, msatPerUnit, erin.PubKeyStr,
		quote.AcceptedQuote.Scid)
```

Reply:
Yeah, I generally agree. And with the new `slog` based logging library we'll be able to do that in a more generalized manner (at least in `lnd`).
For now I'm trying to keep this aligned with other such messages in other tests.